### PR TITLE
NEW Color picker field

### PR DESCRIPTION
**Pull request by scott1702**
Just a PR to cleanly see all the changes made by everyone before progressing the changes forward to the upstream PR

### CODE REVIEW

File: client/src/components/ColorPickerField/ColorPickerField.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        id={`Popover_${name}`}
+        onClick={this.handleToggle}
+        className="color-picker-field-button"
+      >
+        <div
+          className="color-picker-field-button__color-icon"
+          style={{ backgroundColor: color ? color.Color : 'transparent' }}
+        />
+        <div className="color-picker-field-button__color-label">
+          { color ? color.Title : <em>None</em> }
```
Comment (scott1702) [inline client/src/components/ColorPickerField/ColorPickerField.js]: translate 'None'

File: client/src/components/ColorPickerField/ColorPickerField.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    }));
+
+    const handleSearch = (term, set) => set.filter(
+      ({ text }) => text.toLowerCase().includes(term.toLowerCase())
+    );
+
+    return (
+      <PopoverOptionSetComponent
+        buttons={buttons}
+        searchPlaceholder={i18n._t('ColorPickerField.SEARCH_BLOCKS', 'Search colors')}
```
Comment (scott1702) [inline client/src/components/ColorPickerField/ColorPickerField.js]: Change default to british english

File: client/src/components/ColorPickerField/ColorPickerField.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        this.handleToggle();
+        this.setState({
+          value: color.CSSClass,
+        });
+      },
+    }));
+
+    const handleSearch = (term, set) => set.filter(
+      ({ text }) => text.toLowerCase().includes(term.toLowerCase())
+    );
```
Comment (scott1702) [inline client/src/components/ColorPickerField/ColorPickerField.js:82]: Should search be disabled? check with James

File: client/src/components/FontPickerField/FontPickerField.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  renderSelectorButton() {
+    const { value } = this.state;
+    const { fonts, name } = this.props;
+    let font;
+
+    if (value) {
+      font = fonts.find(({ CSSClass }) => CSSClass === value);
+    }
+    if (!font) {
+      font = fonts[0];
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.js]: `font = font || fonts[0]`

Comment (scott1702) [reply client/src/components/FontPickerField/FontPickerField.js]: Apply to ColorPickerField too

File: client/src/components/FontPickerField/FontPickerField.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    return (
+      <Button
+        id={`Popover_${name}`}
+        onClick={this.handleToggle}
+        className="font-picker-field-button font-icon-caret-up-down"
+      >
+        { font ? font.Title : <em>None</em> }
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.js]: Translate

File: client/src/components/FontPickerField/FontPickerField.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      className: 'font-picker-field-popover__option',
+      buttonProps: { style: { fontFamily: `"${font.Title}"` } },
+      text: font.Title,
+    }));
+
+    return (
+      <PopoverOptionSetComponent
+        buttons={buttons}
+        provideButtonClickHandler={this.handleButtonClick}
+        searchPlaceholder={i18n._t('FontPickerField.SEARCH_BLOCKS', 'Search font families')}
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.js]: disableSearch for this dropdown

File: client/src/components/FontPickerField/FontPickerField.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        toggle={this.handleToggle}
+      />
+    );
+  }
+
+  renderSelectedFontPreview() {
+    const { selectedFont } = this.state;
+    const previewText = i18n._t(
+      'FontPickerField.PREVIEW_FONT',
+      'The quick brown fox jumped over the lazy dog'
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.js]: See https://github.com/spaceship-operators/cwp-agencyextensions/pull/3#issuecomment-433746817

File: client/src/components/FontPickerField/FontPickerField.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    );
+    return (
+      <div
+        className="font-picker-field__selection-preview"
+        style={{ fontFamily: `"${selectedFont}"` || 'inherit' }}
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.js]: Make quotes consistent

File: client/src/components/FontPickerField/FontPickerField.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+      content: font.Title,
+      className: 'font-picker-field-popover__option',
+      buttonProps: { style: { fontFamily: `"${font.Title}"` } },
+      text: font.Title,
+    }));
+
+    return (
+      <PopoverOptionSetComponent
+        buttons={buttons}
+        provideButtonClickHandler={this.handleButtonClick}
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.js]: Note this prop has changed, when updating also ensure default is updated https://github.com/spaceship-operators/cwp-agencyextensions/pull/3#issuecomment-433745455

File: client/src/components/FontPickerField/FontPickerField.scss [hunk 10]
```diff
@@ -0,0 +1,39 @@
+.font-picker-field-button {
+  background-color: white;
+  border: 1px solid $border-color-light;
+  min-width: 270px;
+  width: 100%;
+  display: flex;
+  padding-top: $input-padding-y;
+  padding-bottom: $input-padding-y;
+
+  &[class*="font-icon-caret-up-down"]::before {
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.scss]: Check if needed

File: client/src/components/FontPickerField/FontPickerField.scss [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  &__option {
+    padding: 4px;
+    max-height: 50px;
+    width: 100%;
+
+    &:nth-child(odd) {
+      margin-right: 0;
+    }
+
+    &:nth-child(-n+2) {
```
Comment (scott1702) [inline client/src/components/FontPickerField/FontPickerField.scss]: Check if needed after search is disabled

File: client/src/styles/bundle.scss [hunk 12]
```diff
@@ -0,0 +1,10 @@
+// Core variables
+@import 'variables';
+
+// Thirdparty
+@import '~bootstrap/scss/functions';
+@import '~bootstrap/scss/mixins';
+@import '~bootstrap/scss/variables';
```
Comment (scott1702) [inline client/src/styles/bundle.scss]: Check if 3rd party dependencies are needed (probably yes for BS variables)

File: client/src/styles/bundle.scss [hunk 13]
```diff
@@ -0,0 +1,10 @@
+// Core variables
+@import 'variables';
```
Comment (scott1702) [inline client/src/styles/bundle.scss:2]: See if this does anything, can't see variables.scss in agencyextensions

File: src/Extensions/CWPSiteConfigExtension.php [hunk 14]
```diff
@@ -2,6 +2,8 @@
 
 namespace CWP\AgencyExtensions\Extensions;
 
+use CWP\AgencyExtensions\Forms\ColorPickerField;
+use CWP\AgencyExtensions\Forms\FontPickerField;
 use Heyday\ColorPalette\Fields\ColorPaletteField;
```
Comment (scott1702) [inline src/Extensions/CWPSiteConfigExtension.php]: Remove this and any suggested dependency from composer.json

**Comment by scott1702** [comment]
Update docs as well

File: client/src/components/ColorPickerField/ColorPickerField.js (hunk 3 above)
Comment (scott1702) [reply client/src/components/ColorPickerField/ColorPickerField.js:82]: James says keep search 👍

File: client/src/styles/bundle.scss (hunk 13 above)
Comment (scott1702) [reply client/src/styles/bundle.scss:2]: Pulls from silverstripe-admin module due to silverstripe/webpack-config

File: client/src/styles/bundle.scss (hunk 12 above)
Comment (scott1702) [reply client/src/styles/bundle.scss]: Apparently not

File: client/src/styles/bundle.scss (hunk 13 above)
Comment (ScopeyNZ) [reply client/src/styles/bundle.scss:2]: Ah yep - this is pretty common in some other modules for pulling default variables for admin. Perhaps we should comment this import in our modules 🤔

**Pull request merged**